packages/jest/src/automatic.ts

```diff
@@ -34,33 +34,34 @@ const defaultAutoCheckOpts: AutoCheckOpts = {
 };
 
 /**
- * Run accessibility check on each element node in the body using {@link toBeAccessible}
- * @param opts - Options for automatic checks {@link AutoCheckOpts}
+ * Check if current test file needs to be skipped based on any provided filter
  */
-export async function automaticCheck(opts: AutoCheckOpts = defaultAutoCheckOpts): Promise<void> {
-    const violations: AxeResults = [];
-    const testPath = expect.getState().testPath;
-    // If option to run only on selected files is specified, then current path should
-    // match against at least one file
-    if (
-        opts.filesFilter?.length &&
-        !opts.filesFilter
-            // Convenience shortcut converting '!' prefix to filename
-            // as negative lookahead assertion regex
-            .map((filename) => (filename.startsWith('!') ? `/(?!${filename})$` : filename))
-            .some((fileName) => RegExp(fileName).test(testPath))
-    ) {
+export function skipTest(testPath: string, filesFilter?: string[]): boolean {
+    if (!filesFilter || !(filesFilter?.length > 0)) return false;
+    const skipTest = filesFilter.some((fileName) => RegExp(fileName).test(testPath));
```

Comment:
Do we expect the filesFilter entries to be the full file paths? If so, it's probably better to do a string comparison (if not === then a startsWith or contains or similar) instead of using a regex. I see how regex made sense before since we let them use the `!` negation, but to only exclude specific file paths I think a full regex is a little heavy handed.